src/realm/sync/tools/apply_to_state_command.cpp

```diff
@@ -294,11 +303,21 @@ int main(int argc, const char** argv)
         mpark::visit(realm::util::overload{
                          [&](const DownloadMessage& download_message) {
                              realm::sync::VersionInfo version_info;
-                             auto transact = bool(flx_sync_arg) ? local_db->start_write() : local_db->start_read();
+                             TransactionRef tr;
+                             switch (download_message.batch_state) {
+                                 case realm::sync::DownloadBatchState::SteadyState:
```

Comment:
Is it OK that these changes aren't backwards compatible with sync messages from before sync protocol version 14? I am assuming so, since the `history.integrate_server_changesets()` function isn't going to work with older messages since it is explicitly looking for the `DownloadBatchState::SteadyState` value that was introduced in version 14.

Perhaps we should include the `get_current_protocol_version()` output in the help text to indicate which protocol version  "should" be supported?